blockchain/message_pool/src/block_prob.rs

```diff
@@ -0,0 +1,124 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use statrs::function::gamma::ln_gamma;
+use std::f64::consts::E;
+
+const MAX_BLOCKS: usize = 15;
+const MU: f64 = 5.0;
+
+fn poiss_pdf(x: f64, mu: f64, cond: f64) -> f64 {
+    let ln_gamma = ln_gamma(x + 1.0);
+    let log_mu = mu.log(E);
+    let exponent = log_mu * x - ln_gamma - cond;
+    E.powf(exponent)
+}
+
+pub fn no_winners_prob() -> Vec<f64> {
+    let mut out: Vec<f64> = vec![];
+    for i in 0..MAX_BLOCKS {
```

Comment:
either make this vector have a capacity, or iterate and collect based on i, this is inefficient